pkg/cloud/aws/actuators/cluster/actuator.go

```diff
@@ -88,6 +97,28 @@ func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) error {
 		return errors.Wrapf(err, "failed to reconcile load balancers for cluster %q", cluster.Name)
 	}
 
+	// Store KubeConfig for Cluster API NodeRef controller to use.
+	kubeConfig, err := a.Deployer.GetKubeConfig(cluster, nil)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get kubeconfig for cluster %q", cluster.Name)
+	}
+
+	kubeConfigSecret := &apiv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: remote.KubeConfigSecretName(cluster.Name),
+		},
+		StringData: map[string]string{
+			"value": kubeConfig,
+		},
+	}
+
+	if _, err := a.coreClient.Secrets(cluster.Namespace).Create(kubeConfigSecret); err != nil {
```

Comment:
I would add a `Get()` check to see if it exists, and only try to create if it doesn't.